mobile/library/common/config/config.cc

```diff
@@ -267,7 +267,6 @@ const char* config_template = R"(
             - name: remote_service
               domains: ["*"]
               routes:
-#{fake_remote_responses}
```

Comment:
oh, I thought that we would be able to remove a bigger portion of this fake listener as part of this step. What's preventing us from doing that (apologies if you have explained it in the community sync meeting)?

Reply:
I was going to do that as a follow-up after I removed listener support so we can make sure it doesn't break Lyft tests first :-)

Reply:
so the plan is land this, then https://github.com/envoyproxy/envoy/pull/24363 (working on a c-opt for legacy tests like the kotlin tests) then remove the listener entirely